app/javascript/angular/code/services/google/_map.js

```diff
@@ -149,15 +151,35 @@ export const map = (
       return newMarker;
     },
 
-    drawCustomMarker: function({ latLng, content, colorClass, callback, type }) {
-      const customMarker = buildCustomMarker(latLng, content, colorClass, callback, type);
+    drawCustomMarker: function({ latLng, content, colorClass, callback, type, objectId, value }) {
+      const customMarker = buildCustomMarker(latLng, content, colorClass, callback, type, objectId, value );
 
       customMarker.setMap(this.get());
       this.markers.push(customMarker);
 
       return customMarker;
     },
 
+    clusterMarkers: function(onClick) {
+      const options = {
+        styles: [
+          { url: '/assets/marker1.png', height: 10, width: 10 },
+          { url: '/assets/marker2.png', height: 10, width: 10 },
+          { url: '/assets/marker3.png', height: 10, width: 10 },
+          { url: '/assets/marker4.png', height: 10, width: 10 },
+        ],
+        zoomOnClick: false,
+        calculator: (markers, stylesCount) => {
+          const average = markers.reduce((sum, marker) => sum + marker.value(), 0) / markers.length
+          return { text: "", index: heat.getLevel(Math.round(average)) }
+        }
+      };
```

Comment:
I guess it's not easy to understand how this works. In short the index returned by the calculator is used to choose a style from the styles array. It's documented here: https://htmlpreview.github.io/?https://github.com/googlemaps/v3-utility-library/blob/master/markerclustererplus/docs/reference.html#ClusterIconStyle

Reply:
I guess a comment in code would be a good first step